pootle/apps/pootle_store/migrations/0027_unit_created_by_squashed_0055_fill_unit_source_data.py

```diff
@@ -0,0 +1,432 @@
+
+
+def create_sources_with_orm(apps, creators):
+    sysuser = get_system_user_id()
+    UnitSource = apps.get_model("pootle_store.UnitSource")
+    units = apps.get_model("pootle_store.Unit").objects.all()
+    _units = list(
+        units.values_list(
+            "id",
+            "source_hash",
+            "source_length",
+            "source_wordcount").iterator())
+
+    def _unit_source_create(pk, source_hash, source_length, source_wordcount):
+        return dict(
+            unit_id=pk,
+            source_hash=source_hash,
+            source_wordcount=source_wordcount,
+            source_length=source_length,
+            created_with=SubmissionTypes.WEB,
+            created_by_id=creators.get(pk, sysuser))
+    Batch(UnitSource.objects, batch_size=500).create(
+        _units,
+        _unit_source_create,
+        reduces=False)
+
+
+def update_sources_with_orm(apps, creators):
+    UnitSource = apps.get_model("pootle_store.UnitSource")
+
+    def _set_created_by(unit_source):
+        unit_source.created_by_id = creators.get(unit_source.unit_id)
+        return unit_source
+    Batch(UnitSource.objects, batch_size=500).update(
+        list(UnitSource.objects.filter(unit_id__in=creators.keys())),
+        update_method=_set_created_by,
+        update_fields=["created_by"],
+        reduces=False)
+
+
+def create_changes_with_orm(apps):
+    UnitChange = apps.get_model("pootle_store.UnitChange")
+    missing = list(
+        _missing_changes(apps).iterator())
+
+    def _unit_change_create(unit):
+        return dict(
+            unit_id=unit.pk,
+            changed_with=SubmissionTypes.WEB,
+            commented_by_id=unit.commented_by_id,
+            commented_on=unit.commented_on,
+            reviewed_by_id=unit.reviewed_by_id,
+            reviewed_on=unit.reviewed_on,
+            submitted_by_id=unit.submitted_by_id,
+            submitted_on=unit.submitted_on)
+    Batch(UnitChange.objects, batch_size=500).create(
+        missing,
+        _unit_change_create,
+        reduces=False)
+
+
+def create_sources_with_sql(schema_editor):
+    sysuser = get_system_user_id()
+    cursor = schema_editor.connection.cursor()
+    cursor.execute(UNIT_SOURCE_SQL % (SubmissionTypes.SYSTEM, sysuser))
+
+
+def create_changes_with_sql(schema_editor):
+    cursor = schema_editor.connection.cursor()
+    cursor.execute(UNIT_CHANGE_SQL % SubmissionTypes.WEB)
+
+
+def add_unit_sources(apps, schema_editor):
+    subs = apps.get_model("pootle_statistics.Submission").objects.all()
+    sysuser = get_system_user_id()
+    # type 10 is the now deleted UNIT_CREATE
+    creators = dict(
+        subs.filter(type=10)
+            .exclude(submitter_id=sysuser)
+            .values_list("unit_id", "submitter"))
+    if schema_editor.connection.vendor == "mysql" and settings.POOTLE_SQL_MIGRATIONS:
+        create_sources_with_sql(schema_editor)
+        update_sources_with_orm(apps, creators)
```

Comment:
`update_sources_with_orm` shouldn't be necessary if `create_sources_with_sql` uses the value for `sysuser` PK instead of getting it from `pootle_store_unit` table in line 27: https://github.com/translate/pootle/pull/6377/commits/76b79123ee6f1cbbd88b8aaf6db9625af9cd38d4#diff-2912b58a9848b8ae11b5fe7d0bc0da2fR27

That would mean less queries to update the data. Isn't that possible?